examples/repl/repl.cpp

```diff
@@ -0,0 +1,60 @@
+#include <kalcy/kalcy.hpp>
+#include <cassert>
+#include <format>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace {
+struct Repl {
+	const std::string_view EXIT_TOKEN{"exit"};
+	bool verbose{}; // toggled on with '-v' and off with '-q'
+	bool is_running{true};
+
+	auto start() -> bool {
+		while (is_running) {
+			std::string text{};
+			std::cout << std::format("[{}] > ", verbose ? "verbose" : "quiet");
+			std::getline(std::cin, text);
+			// run kalcy on input expression
+			if (!run(text)) { return EXIT_FAILURE; }
+		}
+		// print epilogue
+		std::cout << std::format("\n^^ kalcy v{}\n", kalcy::version_v);
+		return EXIT_SUCCESS;
+	}
+
+	auto run(std::string_view const text) -> bool {
+		try {
+			// return false when the user enters the exit token
+			if (text == EXIT_TOKEN) {
+				is_running = false;
+			} else if (text == "-v") {
+				verbose = !verbose;
+			}  else {
+				// parse text into an expression
+				auto expr = kalcy::Parser{}.parse(text);
```

Comment:
It makes sense to have a member `Parser` instead of creating a fresh one every time.